frontend/purposeful-frontend/components/Registration.js

```diff
@@ -14,11 +14,93 @@ import {
 } from "@chakra-ui/react";
 import { ViewIcon, ViewOffIcon } from "@chakra-ui/icons";
 import { Field, Form, Formik } from "formik";
+import fetchWrapper, { getAuthorities } from "../utils/fetch_wrapper";
+import notification from "../utils/notification";
+
+async function handleRegistrationForm(values, actions, GrantedAuth) {
+  const payload = {
+    email: values.email,
+    password: values.password,
+    firstname: values.firstname,
+    lastname: values.lastname,
+  };
+  let response = null;
+
+  if (
+    GrantedAuth.length === 0 &&
+    !window.location.pathname.includes("moderator")
+  ) {
+    // User is not authenticated and is registering as a regular user
```

Comment:
So we are assuming that if someone is not logged in, and they are not on the register moderator page, they are registering a new account. Correct :)

Reply:
Correct. If they are not logged in, they shouldn't even have access to the register moderator page. We will try to restrict the page endpoint next week